syntax = "proto3";

package kvcache;

// Service definition for performing inference.
service KVCacheService {
  // Inference RPC takes a probe and an optional kv_cache blob.
  rpc Inference(InferenceRequest) returns (InferenceResponse) {}
}

// Request message for inference.
message InferenceRequest {
  string input = 1;
  bytes kv_cache = 2; // Optional: if provided, the server will use this kv cache.
  int32 prompt_len = 3;
}

// Response message containing the first token.
message InferenceResponse {
  string first_token = 1;
}
